src/lib.rs

```diff
@@ -274,6 +287,32 @@ impl PrometheusMetrics {
         })
     }
 
+    fn register_metrics(registry: &Registry, namespace: &str) -> prometheus::Result<(IntCounterVec, HistogramVec)> {
+        let http_requests_total_opts = opts!(
+            "http_requests_total", "Total number of HTTP requests"
+        ).namespace(namespace);
+        let http_requests_total = IntCounterVec::new(
+            http_requests_total_opts,
+            &["endpoint", "method", "status"],
+        ).unwrap();
+
+        registry.register(Box::new(http_requests_total.clone()))?;
+
+        let http_requests_duration_seconds_opts = opts!(
+            "http_requests_duration_seconds",
+            "HTTP request duration in seconds for all requests"
+        )
+        .namespace(namespace);
+        let http_requests_duration_seconds = HistogramVec::new(
+            http_requests_duration_seconds_opts.into(),
+            &["endpoint", "method", "status"],
+        ).unwrap();
+
+        registry.register(Box::new(http_requests_duration_seconds.clone()))?;
+
+        Ok((http_requests_total, http_requests_duration_seconds))
+    }
+
```

Comment:
Ideally a cfg attribute on the parameter would make duplicating this method unnecessary, but that would require rust nightly, since attributes on parameters are not stabilized yet (proposed to be stabilized soon though!)

Reply:
I believe it's now in 1.39 so this can be changed.

Reply:
It requires that we bump the minimum rust version to 1.39, if that's okay we can do that.